agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,277 @@
+
+    /**
+     * Stock agent
+     *
+     * @param client An http client object
+     * @param apiKey A Quandl API key
+     */
+    public StockAgent(HttpClient client, String apiKey) {
+        this.client = client;
+        this.apiKey = apiKey;
+
+        ICsvBeanReader csvBeanReader = null;
+
+        InputStream nasdaqCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NASDAQ_Companies.csv");
+
+        this.stockSymbolMap = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader(new InputStreamReader(nasdaqCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+                this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            logger.debug("Failed to load stockagent/NASDAQ_Companies.csv");
+        }
+
+        InputStream nyseCompanies = StockAgent.class.getClassLoader()
+                .getResourceAsStream("stockagent/NYSE_Companies.csv");
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader(new InputStreamReader(nyseCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+
+                if (!this.stockSymbolMap.containsKey(companyInfo.getName())) {
+                    this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+                }
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            logger.debug("Failed to load stockagent/NYSE_Companies.csv");
+        } 
+
+    }
+
+    @Override
+    public AgentResponse process(AgentRequest request) {
+        logger.debug("Handling the request: " + request.getText());
+
+        String requestCompanyName = request.getText();
+        String[] stockCompanyInfo = findStockSymbol(requestCompanyName);
+        String keywordCompanyName = null;
+        String keywordCompanySymbol = null;
+
+        if (stockCompanyInfo[0] != null) {
+            keywordCompanyName = stockCompanyInfo[0].substring(0, 1).toUpperCase()
+                    + stockCompanyInfo[0].substring(1, stockCompanyInfo[0].length());
+            keywordCompanySymbol = stockCompanyInfo[1];
+        }
+
+        String json = null;
+
+        if (keywordCompanySymbol == null) {
+            return new AgentResponse(
+                    "I am sorry, I don't understand which company you are asking about.");
```

Comment:
Set the response code to ResponseStatus.Code.AGENT_DID_NOT_KNOW_ANSWER. Otherwise, this returns a success.